rib/reconciler/reconcile.go

```diff
@@ -161,7 +168,7 @@ func newReconcileOps() *reconcileOps {
 //
 // If an entry within the explicitReplace map is set to true then explicit, rather
 // than implicit replaces are generated for that function.
-func diff(src, dst *rib.RIB, explicitReplace map[spb.AFTType]bool) (*reconcileOps, error) {
+func diff(src, dst *rib.RIB, explicitReplace map[spb.AFTType]bool, id *atomic.Uint64) (*ReconcileOps, error) {
```

Comment:
Could you explain why an atomic is needed here? Could there be concurrent calls to diff?

Reply:
In an implementation that uses this library that has something like:

 * goroutine A writes updates from an external caller to a `gRIBI.Modify` channel
 * goroutine B calculates reconciliation transactions to be written to the same `gRIBI.Modify` channel

then it's possible that there can be concurrent calls to `Add(1)` or `Load` on the ID, but they need to be coherent with one another -- hence using atomic. An alternative is to have `Reconcile` always return IDs starting at 1, and have the caller rewrite those, but that seems more expensive.